### PR TITLE
add get_asn1_flag to EcGroupRef

### PR DESCRIPTION
**Pull request by reaperhulk**

### CODE REVIEW

**Comment by alex** [comment]
BoringSSL failures are pre-existing, but the OpenSSL ones are related to this patch.

File: openssl/src/ec.rs [hunk 1]
```diff
@@ -294,6 +294,12 @@ impl EcGroupRef {
         }
     }
 
+    /// Gets the flag determining if the group corresponds to a named curve.
+    #[corresponds(EC_GROUP_get_asn1_flag)]
+    pub fn get_asn1_flag(&mut self) -> Asn1Flag {
```
Comment (sfackler) [inline openssl/src/ec.rs:299]: The rustic naming for this would just be `asn1_flag`, without the `get_`.

**Comment by alex** [comment]
LGTM

CI won't be green until https://github.com/sfackler/rust-openssl/pull/1948 is merged

**Comment by alex** [comment]
libressl 2.5.5 also appears sad

File: openssl/src/ec.rs [hunk 2]
```diff
@@ -1265,4 +1271,12 @@ mod test {
         let group2 = EcGroup::from_curve_name(Nid::X9_62_PRIME239V3).unwrap();
         assert!(!g.is_on_curve(&group2, &mut ctx).unwrap());
     }
+
+    #[test]
+    #[cfg(not(any(ossl102, ossl101)))]
+    fn asn1_flag() {
+        let mut group = EcGroup::from_curve_name(Nid::X9_62_PRIME256V1).unwrap();
+        let flag = group.asn1_flag();
+        assert_eq!(flag.0, Asn1Flag::NAMED_CURVE.0);
```
Comment (alex) [inline openssl/src/ec.rs:1280]: Instead of accessing `.0`, can you add a `PartialEq` impl on `Asn1Flag`?

**Review by alex — APPROVED** [review]

**Pull request merged**